### DISCUSSION TITLE
What's the difference between hyper fetch and tanstack?

### DISCUSSION THREAD

**Discussion by Nisthar** [BetterTyped/hyper-fetch#35] [Q&A] [answered]
Tanstack added support for svelte https://tanstack.com/query/latest/docs/svelte/overview 

what are the differences if there are any?

**Comment by endlessuniverse** [comment, marked as answer]
https://hyperfetch.bettertyped.com/docs/documentation/getting-started/comparison/

**Reply by prc5** [reply]
I think this needs to be updated after tons of features we added to it, but more less it's a good comparison. 👍🏻

**Comment by prc5** [comment]
Difference is that we are not wrapper for the state and react/other-environments, we are providing it with the fetchers. 

What this gives us:
- More control over how our request are presented in the application
- Simpler/faster setup - you don't have to know fetch, xhr at all - you initialize your client and request and you are ready to go!
- Websockets/SSE and in general realtime connection module with integrations to make it easier for building realtime apps
- More typescript options and helpers - this gives us a lot of control
- Better reusibility since we made the Requests commands as a part of the solutions
- Upload/Download progress tracking is simpler
- Better readability of the code thanks to the helper hooks like `onSuccess((data) => console.log(data))`

There are more up on the list presented by the @endlessuniverse

**If I have to specify why I build the HF since the React Query is such a great solution is:**

I made this because I wanted to make the data exchange more structured, as React query follows the path of letting user to specify how this data should be exchanged with the server, I simply wanted to do it once for all. I had many problems with upload/download progress, queueing my requests and I just had to write it on my own due to the complex requirements of my products at that time. Right now I know that this was brilliant idea as it brings standards to the way of how you manage your data and how you create reusable parts for the data exchange. At every single stage of the request, you can inspect it's transferred data, params, query params, etc. This is simply going beyond of what's possible with other solutions, together with typescript I'm using it for years and I never looked back, it's just simple and powerful.
